src/uthash.h

```diff
@@ -245,12 +245,14 @@ do {
     struct UT_hash_handle *_hs_iter = &(head)->hh;                               \
     (add)->hh.tbl = (head)->hh.tbl;                                              \
     do {                                                                         \
-      if (cmpfcn(DECLTYPE(head) ELMT_FROM_HH((head)->hh.tbl, _hs_iter), add) > 0) \
+      if (cmpfcn(DECLTYPE(head) ELMT_FROM_HH((head)->hh.tbl, _hs_iter), add) > 0) { \
         break;                                                                   \
-    } while ((_hs_iter = _hs_iter->next));                                       \
-    if (_hs_iter) {                                                              \
+      }                                                                          \
+    } while ((_hs_iter = _hs_iter->next) != NULL);                               \
+    if (_hs_iter != NULL) {                                                      \
       (add)->hh.next = _hs_iter;                                                 \
-      if (((add)->hh.prev = _hs_iter->prev)) {                                   \
+      (add)->hh.prev = _hs_iter->prev;                                           \
+      if ((add)->hh.prev != NULL) {                                              \
```

Comment:
I'm amenable to this change, although it doesn't seem particularly useful to me. I'd prefer a patch with a commit message such as "uthash produces the following diagnostics on platform Foo: (bar)... Suppress these diagnostics by consistently (baz)..."

And if that patch could consist entirely of `#ifdef Foo ... #pragma warning suppress bar ... #endif` somewhere at the top of this header, then that would be even better, because it wouldn't penalize anybody who didn't use platform Foo.